### PR TITLE
More php7 improvements

### PR DESCRIPTION
**Pull request by Klap-in**

### CODE REVIEW

File: inc/feedcreator.class.php [hunk 1]
```diff
@@ -505,7 +510,7 @@ function addItem($item) {
      * @param int     $length the maximum length the string should be truncated to
      * @return string    the truncated string
      */
-    function iTrunc($string, $length) {
+    static function iTrunc($string, $length) {
```
Comment (splitbrain) [inline inc/feedcreator.class.php]: why the change to static?

File: inc/cliopts.php [hunk 2]
```diff
@@ -447,7 +447,7 @@ class Doku_Cli_Opts_Error {
     var $code;
     var $msg;
 
-    function Doku_Cli_Opts_Error($code, $msg) {
+    function __construct($code, $msg) {
```
Comment (splitbrain) [inline inc/cliopts.php]: the change is fine, but the class itself is deprecated anyway, so maybe we can remove it (have to check the date)

Comment (Klap-in) [inline inc/cliopts.php]: Removing deprecated stuff is a different job, good for other commit.

File: inc/feedcreator.class.php (hunk 1 above)
Comment (Klap-in) [inline inc/feedcreator.class.php]: It is an annotation in the PHPDocs, it is also suggested by IntelliJ IDEA.
Further this function is indeed everywhere called as static function in DokuWiki.

**Comment by Klap-in** [comment]
This is fine to merge?
